libs/server/repository/api-rest/src/lib/controllers/repository.controller.ts

```diff
@@ -39,16 +50,23 @@ export class RepositoryController {
     @Credentials() credentials: RequestCredentials,
     @CurrentUserId() userId: string
   ): Promise<SingleRepositoryDataReadModel> {
-    return this.repositoryFacade.addRepository(
-      new AddRepositoryCommand(
-        extractFullName(addRepositoryDto.repositoryUrl),
-        userId,
-        credentials.token,
-        credentials.platform,
-        addRepositoryDto.maxLines,
-        addRepositoryDto.maxWaitingTime
-      )
-    );
+    const fullName = extractFullName(addRepositoryDto.repositoryUrl);
+    return this.repositoryFacade.checkIfRepositoryHasBeenAdded(userId, fullName).then(exists => {
+      if (exists) {
```

Comment:
It is like business logic, and nothing of that should be in the controller. Move it deeper.